recipes/jerryscript/all/conanfile.py

```diff
@@ -41,7 +115,36 @@ def configure(self):
         del self.settings.compiler.cppstd
         if self.options.shared:
             del self.options.fPIC
-            
+
+        if not self.options.debugger:
+            del self.options.keep_line_info
+        
+        if self.options.profile is None:
+            if Version(self.version) < Version("2.4.0"):
+                self.options.profile = "es5.1"
+            else:
+                self.options.profile = "es.next"
+        if (not self.options.profile in self.predefined_profiles) and not os.path.isfile(self.options.profile):
+            raise ConanInvalidConfiguration("jerryscript feature profile must either be a valid file or one of these: es.next, es5.1, minimal")
+        
+        try:
+            assert(int(self.options.heap_size) >= 0), "must be bigger than or equal to 0"
+            assert(int(self.options.gc_mark_limit) >= 0), "must be bigger than or equal to 0"
+            assert(int(self.options.stack_limit) >= 0), "must be bigger than or equal to 0"
+        except:
+            raise ConanInvalidConfiguration("jerryscript heap size, gc mark limit, stack limit should be a positive integer")
+
+        if self.options.jerry_math is None:
```

Comment:
Move this logic to `config_options` and remove the tests in `configure`.
`config_options` can be used to modify default options for certain platforms/versions.
```python
def config_options(self):
    if Version(self.version) < "2.4.0":  # No need to use tools.Version twice
        self.options.jerry_math = True
    else:
        self.options.jerry_match = False
```